feat(kno-4774): add example app branch without knock

meryldakin opened the pull request:
This adds a branch of this project that should remain unmerged and can be used to practice adding Knock into the example app, both by Knock users and new Knock team members.

connorlindsey reviewed (commented):
I'm curious where we want to land in terms of how much instruction we give. The original app only provided [high-level instructions](https://github.com/knocklabs/collab-onboarding-app/blob/main/docs/01_goals.md) which forces you to search more in the docs. This is much more guided. I think if the goal is to test out the product like a new customer would, we should lean more towards the hands-off approach by giving less instruction here and maybe removing the explicit TODO: Add Knock comments. 

It is a balance though because we don't want the exercise to feel tedious. Normally a customer would already be familiar with the code base they're implementing Knock in, but in this case there are 2 unknowns (Knock and the demo app), so we should provide some guidance on how the app is architected so that the focus is on learning Knock. I think TODO comments can work there or we could provide an architecture.md that walks through the high level architecture of the app and points out relevant files. 

A minor discussion point: how do we feel about this only using JavaScript vs. the other example using Elixir and JS? I think it'd be nice to show multiple SDKs but feels out of scope and not worth the squeeze.

meryldakin commented:
That's a really good question and totally up for debate. I made it extremely clear, I guess because 1) it's easier to pull back from what we have here to something more vague, and 2) this is not engineering onboarding.

I think however this first round goes we'll probably want to amend it based on their feedback. Maybe it's better to strip it down more from this and if it ends up being too hard or confusing, add instructions in for the next set of hires. I was a little worried about the thing you're bringing up where this is a built out, unfamiliar app, and Knock is also unfamiliar, so hand-holding would be fine for them to get the hang of it. Maybe one thing we could do is be explicit about the "level 1" requirements but less instructive about the rest (or vice versa)?

vlymar commented:
This is a really interesting idea and I'm curious to hear feedback from new hires and customers who try this.

Some questions:
- when would a customer want to practice adding Knock into our example app? would they prefer to just add it directly into their own app rather than add it 2x? 
- how do you envision us letting customers and new hires know this branch exists? How should they interact with this branch? E.g. should they make a branch off of this branch? should they view it in the github compare view?
- how do we make sure this branch gets kept up to date as we update `main`? `main` is probably updated infrequently, but that makes it much more likely the updater forgets to update this branch as well

meryldakin commented:
> * when would a customer want to practice adding Knock into our example app? would they prefer to just add it directly into their own app rather than add it 2x?

I imagine this happens more when a potential customer is evaluating Knock, or a potential hire is trying to understand our product.

> * how do you envision us letting customers and new hires know this branch exists? How should they interact with this branch? E.g. should they make a branch off of this branch? should they view it in the github compare view?

I'm imagining this being a totally local project for anyone who uses it and restrict pushing to it. So if they don't make their own branch, they would just switch to using this one locally and not make any upstream changes. In terms of how to let them know it exists, I think it's most effective for new hires so I don't feel like we need to widely publicize it, but i was going to add a message about it in the README of main. 

> * how do we make sure this branch gets kept up to date as we update `main`? `main` is probably updated infrequently, but that makes it much more likely the updater forgets to update this branch as well

Good question. I think the best answer I have is making a note in the pr template of this project to port any changes to main to this branch as well, but open to ideas!

meryldakin commented:
> Yeah, those are good points. One option is to progressively give more detail if the person onboarding wants it, so the first readme just has the high level goals and then we have a additional linked pages with more detailed instructions ("If you're stuck, here are some hints…").
@connorlindsey, do you think I should take out the pointers throughout the files too? or make them less explicit? I shaved down the instructions entirely and I'll put them elsewhere and link to it here when it's ready.

connorlindsey commented on app/auth/mutations/signup.ts:
```diff
@@ -33,52 +29,15 @@ export default resolver.pipe(resolver.zod(Signup), async ({ email, password, nam
-      notify.success = false
-      console.error("Error setting preferences:", error)
-    }
+    1. Identify user on Knock so we the data is accessible when triggering workflows
```
Nit: remove 'we'

connorlindsey commented:
@meryldakin I think we should leave them. Without them, I think it'll be a pain trying to find the right places to add the Knock calls. 

Also, what do you think about linking the docs from the readme and pushing users there? I think an important part of the onboarding exercise is testing our docs/in-app onboarding.

meryldakin commented on README.md:
```diff
@@ -70,3 +70,27 @@ to your local web server.
+
+This branch of the Knock example app doesn't have Knock added yet. Throughout the files, you'll see the logical spaces to add calls marked with the comment `TODO: ADD KNOCK`. When added, this application will be able to do the following:
+
+1. [Identify users on Knock](https://docs.knock.app/managing-recipients/identifying-recipients)
```
added links

connorlindsey reviewed (approved):
Thanks for putting this together, I think it looks good!

meryldakin commented:
@connorlindsey @cjbell made a couple of changes here after chatting with chris about it:
1. Removed the `toggle_muted` action to remove complexity in setting preferences
2. Removed the slack integration
3. Remove mentions of segment
4. Added instructions to inline-identify when triggering workflows

This should make the project more streamlined, lmk what you think!

meryldakin commented:
Closing because we don't intend to merge this branch.
